auth.go

```diff
@@ -81,6 +84,44 @@ func getServerPubKey(name string) (pubKey *rsa.PublicKey) {
 	return
 }
 
+// CredentialProviderFunc is a function which can be used to fetch a username/password
+// pair for use when opening a new MySQL connection. The first return value is the username
+// and the second the password.
+type CredentialProviderFunc func() (string, string, error)
```

Comment:
For the purposes of documentation could it be useful to name the return values?

```suggestion
type CredentialProviderFunc func() (username string, password string, err error)
```

Reply:
Agreed, this makes the purpose of the return values clearer, I have updated it.